cdap-data-fabric-tests/src/test/java/co/cask/cdap/data2/transaction/stream/StreamAdminTest.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.data2.transaction.stream;
+
+import co.cask.cdap.api.flow.flowlet.StreamEvent;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.data.file.FileWriter;
+import co.cask.cdap.data.stream.StreamFileWriterFactory;
+import co.cask.cdap.proto.Id;
+import com.google.common.base.Charsets;
+import com.google.common.collect.Lists;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ *
+ */
+public abstract class StreamAdminTest {
+
+  protected abstract StreamAdmin getStreamAdmin();
+
+  protected abstract StreamFileWriterFactory getFileWriterFactory();
+
+  @Test
+  public void testCreateExist() throws Exception {
+    StreamAdmin streamAdmin = getStreamAdmin();
+
+    String streamName = "streamName";
+    Id.Stream streamId = Id.Stream.from(Constants.DEFAULT_NAMESPACE, streamName);
```

Comment:
Possible to use non-default namespaces? The lifecycle of ``DEFAULT_NAMESPACE`` is slightly unpredictable because we never delete it, can cause unpredictable state between tests. Better to create and delete namespaces before and after test/test classes.